arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -1680,25 +1708,28 @@ std::vector<std::string> RocksDBEngine::currentWalFiles() const {
   return names;
 }
 
-Result RocksDBEngine::flushWal(bool waitForSync, bool waitForCollector,
-                               bool /*writeShutdownFile*/) {
+Result RocksDBEngine::flushWal(bool waitForSync, bool waitForCollector) {
+  Result res;
+
   if (_syncThread) {
     // _syncThread may be a nullptr, in case automatic syncing is turned off
-    _syncThread->syncWal();
+    res = _syncThread->syncWal();
   }
 
-  if (waitForCollector) {
+  if (res.ok() && waitForCollector) {
     rocksdb::FlushOptions flushOptions;
     flushOptions.wait = waitForSync;
```

Comment:
It looks like we typically call this function as `flushWal(true, false)`. Except we don't use the first parameter unless the second one is true. Odd.